validation-test/compiler_crashers_2_fixed/0134-rdar35947198.swift

```diff
@@ -0,0 +1,43 @@
+// RUN: %target-swift-frontend %s -emit-ir -o -
```

Comment:
Does this need -O? Also perhaps it should go into test/SILOptimizer instead

Reply:
It does need -O / multiple optimizations to trigger the crash.
There's nothing really interesting, from an optimization POV, happening here:
There's no interesting code pattern to test in test/SILOptimizer, and the compiler is not particularly fast with dealing with this source file. adding it to test would me increasing the smoke test time.

Reply:
But it's the mandatory passes that expose the problem right? I was thinking it would make sense to add some CHECK: lines to ensure the optimizations that should happen happened (otherwise, if the devirtualizer was totally disabled, the test would still pass, but it would not crash without your fix). But it's up to you